aggregator/entry.go

```diff
@@ -211,36 +226,44 @@ func (e *Entry) addMetricWithPoliciesList(
 		return errEntryClosed
 	}
 
-	// Fast exit path for the common case.
-	hasDefaultPoliciesList := pl.IsDefault()
-	if e.hasDefaultPoliciesList && hasDefaultPoliciesList {
-		err := e.addMetricWithLock(currTime, mu)
+	// Fast exit path for the common case where the metric has default metadatas for aggregation.
+	hasDefaultMetadatas := metadatas.IsDefault()
+	if e.hasDefaultMetadatas && hasDefaultMetadatas {
+		err := e.addMetricWithLock(currTime, metric)
 		e.RUnlock()
 		timeLock.RUnlock()
 		return err
 	}
 
-	sp, err := e.activeStagedPoliciesWithLock(pl, currTime)
+	sm, err := e.activeStagedMetadataWithLock(currTime, metadatas)
 	if err != nil {
 		e.RUnlock()
 		timeLock.RUnlock()
 		return err
 	}
 
-	// If the policy indicates the (rollup) metric has been tombstoned, the metric is
+	// If the metadata indicates the (rollup) metric has been tombstoned, the metric is
 	// not ingested for aggregation. However, we do not update the policies asssociated
 	// with this entry and mark it tombstoned because there may be a different raw metric
 	// generating this same (rollup) metric that is actively emitting, meaning this entry
 	// may still be very much alive.
-	if sp.Tombstoned {
+	if sm.Tombstoned {
 		e.RUnlock()
 		timeLock.RUnlock()
-		e.metrics.tombstonedPolicy.Inc(1)
+		e.metrics.tombstonedMetadata.Inc(1)
 		return nil
 	}
 
-	if !e.shouldUpdatePoliciesWithLock(currTime, sp) {
-		err := e.addMetricWithLock(currTime, mu)
+	// It is expected that there is at least one pipeline in the metadata.
+	if len(sm.Pipelines) == 0 {
```

Comment:
Does this mean all the metrics need to carry a pipeline over the network?

Reply:
Yes. Although in most cases it'll be an empty pipeline and the encoder can optimize for that case, but either way it won't be a significant portion of the network usage.